Documentation/test-suites-overview.md

```diff
@@ -0,0 +1,228 @@
+repository](https://github.com/mendersoftware/integration/blob/master/extra/travis-testing/run-test-environment).
+
+An example to run backend acceptance tests for `deviceauth` service follows
+
+```
+# Build the tester
+docker build -f Dockerfile.acceptance-testing -t mendersoftware/deviceauth:prtest .
+# Get wrapper
+git clone --depth=1 https://github.com/mendersoftware/integration.git mender-integration
+cp mender-integration/extra/travis-testing/* tests/
+# Copy swagger specs
+cp docs/* tests/
+# Build and copy the binary
+go build
+cp deviceauth /tests
+# Get and copy mender-artifact
+wget https://downloads.mender.io/mender-artifact/master/linux/mender-artifact
+chmod +x mender-artifact
+cp mender-artifact tests/
+# Run tests!
+TESTS_DIR=$(pwd)/tests $(pwd)/tests/run-test-environment acceptance $(pwd)/mender-integration $(pwd)/tests/docker-compose-acceptance.yml
+```
+
+## client acceptance tests
+
+Test suite       | client acceptance tests
+---------------- | ----------
+ISTQB equivalent | Component integration testing for meta-mender
+Test object      | meta-mender Yocto layer(s) build options
+Source code      | [meta-mender](https://github.com/mendersoftware/meta-mender/tree/master/tests/acceptance) and [mender-image-tests](https://github.com/mendersoftware/mender-image-tests/tree/master/tests) repositories
+How to run       | `python3 -m pytest` (see below)
+Dependencies     | Yocto environment (sourced) with `meta-mender-ci` layer, Python 3, several [Python packages](https://github.com/mendersoftware/meta-mender/blob/master/tests/acceptance/requirements_py3.txt)
+
+This test suite verifies different configuration options, image features and build scenarios for Mender's meta-mender Yocto layer(s). The available tests are repeated for a variety of configurations (see below) and the tests include both static checks and runtime checks.
+
+The platforms under test can be found in [meta-mender
+repository](https://github.com/mendersoftware/meta-mender/tree/master/tests/build-conf).
+A "platform", in this context is a combination of CPU architecture (x86, ARM),
+boot loader integration (UEFI Grub, u-boot, bios) and a file system (ext4,
+ubifs). Follows a list of the current platforms under test.
+
+* qemux86-64-uefi-grub
+* vexpress-qemu
+* vexpress-qemu-flash
+* raspberrypi3
+* raspberrypi4
+* beagleboneblack
+* qemux86-64-bios-grub-gpt
+* qemux86-64-bios-grub
+* vexpress-qemu-uboot-uefi-grub  
+
+### Running locally
+
+To execute the tests, a full Yocto environment (and at least one previously
+build image) is required. A complete recipe of now we build and test this on CI
+can be found at [mender-qa
+repository](https://github.com/mendersoftware/mender-qa/blob/master/scripts/yocto-build-and-test.sh).
+
+The following recipe gives an example to biuld and test `qemux86-64-uefi-grub` platform.
+
+```
+source oe-init-build-env build-qemux86-64-uefi-grub
+bitbake-layers add-layer ../meta-mender/meta-mender-core
```

Comment:
Suggestion:
Maybe comment it up, like you did for the backend tests? Describing each step. That was nice I think

Reply:
Ok, I added:

```
```
# Initialize a new Yocto build environment
source oe-init-build-env build-qemux86-64-uefi-grub
# Add meta-mender build layers
bitbake-layers add-layer ../meta-mender/meta-mender-core
bitbake-layers add-layer ../meta-mender/meta-mender-demo
bitbake-layers add-layer ../meta-mender/meta-mender-qemu
# Add meta-mender test layer
bitbake-layers add-layer ../meta-mender/tests/meta-mender-ci
# Take the default Yocto configuration for this platform
cp ../meta-mender/tests/build-conf/qemux86-64-uefi-grub/local.conf conf/
# Set MENDER_ARTIFACT_NAME
cat >> conf/local.conf <<EOF
MENDER_ARTIFACT_NAME = "mender-image-local"
EOF
# Build full image
bitbake core-image-full-cmdline
# Run the tests!
cd ../meta-mender/tests/acceptance/
python3 -m pytest -p no:xdist --bitbake-image core-image-full-cmdline --board
```

Reply:
Yes, exactly!